src/SFA.DAS.PR.Data/Repositories/AccountProviderLegalEntitiesReadRepository.cs

```diff
@@ -0,0 +1,22 @@
+﻿using Microsoft.EntityFrameworkCore;
+using SFA.DAS.PR.Domain.Entities;
+using SFA.DAS.PR.Domain.Interfaces;
+
+namespace SFA.DAS.PR.Data.Repositories;
+public class AccountProviderLegalEntitiesReadRepository(IProviderRelationshipsDataContext _providerRelationshipsDataContext) : IAccountProviderLegalEntitiesReadRepository
+{
+    public Task<List<AccountProviderLegalEntity>> GetAccountProviderLegalEntities(long? ukprn, string? accountHashId, List<Operation> operations, CancellationToken cancellationToken)
+    {
+        return _providerRelationshipsDataContext.AccountProviderLegalEntities
+            .Include(a => a.AccountProvider)
+                .ThenInclude(a => a.Account)
+            .Include(a => a.Permissions)
+            .Include(a => a.AccountLegalEntity)
+            .Where(t =>
+                (!ukprn.HasValue || t.AccountProvider.ProviderUkprn == ukprn) &&
```

Comment:
`!ukprn.HasValue`  is not going to work out in the query vey well.

Reply:
I see there is validation for Ukprn, so in the repository, the parameter should not be nullable for ukprn and account hash id

Reply:
The logic specifies that either the Ukprn or the PubicHashId must be supplied so either or can be nullable according to 1314